Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+* Upgrading/rebuilding package versions can cause conflicts with other packages
+  not related to mantid and even old mantid versions.
+* New dependencies have to be installed manually on developers and builders machines.
+
+
+### Windows
+
+Windows uses a collection of custom-built libraries managed
+by a homegrown set of [scripts](https://github.com/mantidproject/thirdparty-msvc2015).
+The binaries are stored using the [git lfs](https://git-lfs.github.com/)
+extension for managing large binaries with git.
+For developers, all of the libraries are pulled down by
+the [cmake configure step](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/Bootstrap.cmake#L13)
+thereby minimizing the effort required for developers to get up and running.
+It also includes a bundled version of Python as none is available by default
+on Windows.
+
+The package provided to users contains all required dependencies and is a
+self-contained bundle.
+
+Pros:
+* Easy developer setup.
+* Complete freedom to choose package versions and apply patches etc.
+* New dependencies are retrieved automatically for developers and build servers
+* Self-contained user package.
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Homegrown set of scripts are very clunky.
+* Maintenance of build scripts adds overhead.
+* Complete bundling of dependencies gives a much larger user installer package.
+
+
+### MacOS
+
+MacOS uses the [Homebrew](https://brew.sh/) package manager along with a custom
+set of [formulae](https://github.com/mantidproject/homebrew-mantid) for obselete
+dependencies.
+Homebrew either downloads a prebuilt binary or builds it locally from the formula.
+A developer formula is provided to easily install the required dependencies.
+
+The package provided for users contains all required dependencies
+much like Windows.
+
+Pros:
+* Build formula (mostly) maintained for us.
+* Simply commands to install packages
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* No way to set deployment. We have to build on the earliest OS we intend
+  to support causing issues with hardware availability.
+* New dependencies have to be installed by hand on developers
+  and builders machines.
+* Versions can be pinned but Homebrew really likes to have everything at HEAD
+
+## General Concerns
+
+While each approach has the drawbacks described above
+there are also general issues with the current approach:
+
+1. It is not possible to define a version of a given
+   library as the version used across the whole project.
+   This can cause confusion for developers
+   where their platform might have a newer api
+   yet their code does not work on other platforms.
+   This slows down development.
+2. Using a new library takes a lot of overhead and
+   in some cases is basically impossible
+3. Three different methods requires much additional overhead
+   in understanding how things are managed.
+
```

Comment:
4. Pushing new dependencies to build servers requires co-ordination and potential downtime